funsor/sum_product.py

```diff
@@ -93,6 +93,91 @@ def partial_sum_product(sum_op, prod_op, factors, eliminate=frozenset(), plates=
     return results
 
 
+def modified_partial_sum_product(
+        sum_op, prod_op, factors, eliminate=frozenset(),
+        plate_to_step=dict()):
+    """
+    Modified partial sum-product that supports contraction of markov sites.
```

Comment:
Make the docstring a bit more informative, e.g. "Generalization of the tensor variable elimination algorithm of `funsor.sum_product.partial_sum_product` to handle markov dimensions in addition to plate dimensions.  Markov dimensions are eliminated efficiently using the parallel-scan algorithm in `funsor.sum_product.sequential_sum_product`."